recipes/minizip/all/conanfile.py

```diff
@@ -0,0 +1,138 @@
+                self.settings.os == "watchOS")
+
+    def requirements(self):
+        if self.options.zlib:
+            self.requires("zlib/1.2.11")
+        if self.options.bzip2:
+            self.requires("bzip2/1.0.8")
+        if self.options.zstd:
+            self.requires("zstd/1.4.5")
+        if self.options.openssl:
+            self.requires("openssl/1.1.1g")
+        if self._is_uinix_like():
+            self.requires("libiconv/1.16")
+
+    def configure(self):
+        del self.settings.compiler.libcxx
+        del self.settings.compiler.cppstd
+        if (self.options.signing and
+            not self.options.pkcrypt and
+                not self.options.wzaes):
+            raise ConanInvalidConfiguration(
+                "pkcrypt or wzaes need to be set signing")
+        if self.options.signing and self.options.brg:
+            raise ConanInvalidConfiguration(
+                "Library can not support signing with brg")
+
+    def config_options(self):
+        if self.settings.os == 'Windows':
+            del self.options.fPIC
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        extracted_dir = self.name + "-" + self.version
+        os.rename(extracted_dir, self._source_subfolder)
+
+    def _configure_cmake(self):
+        if not self._cmake:
+            self._cmake = CMake(self)
+            self._cmake.definitions["MZ_COMPAT"] = self.options.compat
+            self._cmake.definitions["MZ_ZLIB"] = self.options.zlib
+            self._cmake.definitions["MZ_BZIP2"] = self.options.bzip2
+            self._cmake.definitions["MZ_LZMA"] = self.options.lzma
+            self._cmake.definitions["MZ_ZSTD"] = self.options.zstd
+            self._cmake.definitions["MZ_PKCRYPT"] = self.options.pkcrypt
+            self._cmake.definitions["MZ_WZAES"] = self.options.wzaes
+            self._cmake.definitions["MZ_LIBCOMP"] = self.options.libcomp
+            self._cmake.definitions["MZ_OPENSSL"] = self.options.openssl
+            self._cmake.definitions["MZ_LIBBSD"] = self.options.libbsd
+            self._cmake.definitions["MZ_BRG"] = self.options.brg
+            self._cmake.definitions["MZ_SIGNING"] = self.options.signing
+            self._cmake.definitions["MZ_COMPRESS_ONLY"] = self.options.compress_only
+            self._cmake.definitions["MZ_DECOMPRESS_ONLY"] = self.options.decompress_only
+            self._cmake.configure(build_folder=self._build_subfolder)
+        return self._cmake
+
+    def build(self):
+        for patch in self.conan_data["patches"][self.version]:
+            tools.patch(**patch)
+        cmake = self._configure_cmake()
+        cmake.build()
+
+    def package(self):
+        self.copy(pattern="LICENSE", dst="licenses",
+                  src=self._source_subfolder)
+        cmake = self._configure_cmake()
+        cmake.install()
+
+    def package_info(self):
+        self.cpp_info.libs = tools.collect_libs(self)
```

Comment:
UNTESTED!
If the package contains multiple libraries, it is important to list them in the correct order.
```suggestion
        self.cpp_info.libs = ["minizip"]
```

Reply:
minizip only generates one library so i would leave it like that.

Reply:
When using minizip as a static library and combining with xz_utils, then you can get weird effects.
Maybe change the option name to `with_lzma`, and add a FIXME about the vendored xz_utils.

Reply:
The library name is listed explicit e129737.
Renamed lzma to with_lzma c665281.
Add FIXME for xz_utils 71d5222.

I also attempted to use xz_utils by [patching](https://github.com/ItsBasi/minizip/tree/conan-patches/2.10.0-xz_utils). While it looked ok, i would delay this change in favor of stabilizing the recipe first.

Reply:
> , i would delay this change in favor of stabilizing the recipe first.

Ok for me.

Reply:
> , i would delay this change in favor of stabilizing the recipe first.

Ok for me.